tiny_dnn/layers/layer.h

```diff
@@ -66,9 +68,13 @@ class layer : public node {
       out_channels_(out_type.size()),
       in_type_(in_type),
       out_type_(out_type) {
+    // todo (karandesai) : remove these after parameter integration
     weight_init_ = std::make_shared<weight_init::xavier>();
     bias_init_   = std::make_shared<weight_init::constant>();
-    trainable_   = true;
+
+    weight_init_f_ = std::make_shared<parameter_init::xavier>();
```

Comment:
not sure if it will brake tests this change. It might be worth adding a TODO for this